src/OSPSuite.Core/Domain/Descriptors/EntityDescriptor.cs

```diff
@@ -0,0 +1,18 @@
+﻿namespace OSPSuite.Core.Domain.Descriptors
+{
+   public class EntityDescriptor
+   {
+      public Tags Tags { get; set; }
+      public Tag Container { get; set; }
+
+      public EntityDescriptor()
+      {
+      }
+
+      public EntityDescriptor(IEntity entity)
+      {
+         Tags = new Tags(entity.Tags){new Tag(entity.Name)};
+         Container = new Tag(entity.ParentContainer?.Name);
```

Comment:
:exclamation: I don't think that this implementation is correct. :exclamation: 

_InContainer-condition_ should check not only the _name_ but also the **_tags_** of the parent container.
(actually _only the tags_, because container-name is always automatically added into the tag-list of every container).

E.g. if a user would like to sum up something related to a group of containers, condition would rely not on the organ name, but on the corresponding tag (total volume of several "tumor" organs for example or whatever else).

Reply:
Actually I'm just asking myself if `EntityDescriptor.Container` should include the tags of **ALL parent containers in the hierarchy of the entity**.

E.g. `EntityDescriptor.Container` of _SmallIntestine/Mucosa/Duodenum/Intracellular/Volume_ would include the tags of 
* SmallIntestine
* SmallIntestine/Mucosa
* SmallIntestine/Mucosa/Duodenum and 
* SmallIntestine/Mucosa/Duodenum/Intracellular

Because that's what is already done in current models quite often: containers become tags of their parent containers in the hierarchy in order to enable e.g. passive transports:
![grafik](https://user-images.githubusercontent.com/25061876/56077363-96c19200-5ddb-11e9-9df4-f703afc91f4d.png)![grafik](https://user-images.githubusercontent.com/25061876/56077382-c53f6d00-5ddb-11e9-9920-72979e643c89.png)


If we implement `EntityDescriptor.Container` with the full hierarchy - there will be no need for this kind of "Tag inheritance" in the future!

Reply:
Interesting. I actually thought about both aspects yesterday as I was implementing it

I agree with the first one. However name would have to be added and not just tags (as you can see, it's added here as well for the entity itself)


I thought about #2 a bit and decided against it because if you want to get something in Liver for example, every sub compartment of liver would also be found as potential container. (You are thinking tree up, I am thinking tree down)

I think it would be very easy to suddenly create stuff (transport, observer) or collect stuff (for the sum Formula) that are not intended

Maybe it's not a big deal. We need to think about it a bit more. (Plus those criteria are created all the time that we check if a condition is satisfied...this could potentially be time consuming!)

Reply:
For the last point, we could delay evaluation of such tags until it's really needed. E.g. No need to retrieve the whole hierarchy for each entity if the condition does not even use "in container"